org.idempiere.test/src/org/idempiere/test/base/MWindowTest.java

```diff
@@ -197,61 +218,53 @@ public void testMandatoryFieldValidation() {
 	}
 
 	/**
-	 * Test creating a persistent AD_Window and cleanup.
-	 * <p>
-	 * Attempts to provide minimal required fields to persist a window. Real DB
-	 * constraints may require additional setup (e.g., Tabs). The test ensures
-	 * cleanup using delete(true, trxName).
+	 * Test creating a persistent AD_Window
 	 */
 	@Test
 	public void testCreatePersistentObjectAndCleanup() {
-		MWindow newWin = new MWindow(ctx, 0, getTrxName());
-		String unique = "T_TEST_WIN_" + System.currentTimeMillis();
 		try {
+			MWindow newWin = new MWindow(ctx, 0, getTrxName());
+			String unique = "T_" + System.currentTimeMillis();
 			newWin.setName("Test Window " + unique);
 			newWin.setWindowType(MWindow.WINDOWTYPE_Maintain);
-			newWin.setIsActive(true);
 			boolean saved = newWin.save();
 			assertTrue(saved, "New AD_Window should save when minimal mandatory fields are provided (Name, WindowType, IsActive)");
 			assertTrue(newWin.get_ID() > 0, "Saved AD_Window must have assigned AD_Window_ID");
 		} finally {
-			try {
-				if (newWin.get_ID() > 0)
-					newWin.delete(true, getTrxName());
-			} catch (Exception e) {
-			}
+			rollback();
 		}
 	}
 
 	/**
-	 * Test helper methods: getWindow_ID(String) and get(ctx, uu) by UUID.
+	 * Test cases for markImmutable()
 	 */
 	@Test
-	public void testHelpers() {
-		int idByName = MWindow.getWindow_ID(testWindowName);
-		assertEquals(testWindowId, idByName, "getWindow_ID(name) should return expected AD_Window_ID");
-
-		MWindow wByUUID = MWindow.get(ctx, testWindowUUID);
-		assertNotNull(wByUUID, "get(ctx, uuid) should return a window instance");
-		assertEquals(testWindowId, wByUUID.getAD_Window_ID(), "Window loaded by UUID must match ID fetched in beforeAll");
-	}
-
-	/**
-	 * Test markImmutable() and setWindowSize(Dimension) do not throw and behave as
-	 * expected.
-	 */
-	@Test
-	public void testMarkImmutableAndSetSize() {
-		MWindow w = new MWindow(ctx, testWindowId, getTrxName());
-		w.markImmutable();
+	public void testMarkImmutable() {
+		MWindow window = new MWindow(ctx, testWindowId, getTrxName());
+		MTab[] tabs = window.getTabs(false, getTrxName());
+		for (MTab tab : tabs)
+			assertFalse(tab.is_Immutable(), "Tab should not be marked as immutable");
+		
+		window.markImmutable();
+		for (MTab tab : tabs)
+			assertTrue(tab.is_Immutable(), "Tab should be marked immutable when window is immutable");
+		
+		tabs = window.getTabs(true, getTrxName());
+		for (MTab tab : tabs)
+			assertTrue(tab.is_Immutable(), "Tab should be marked immutable when window is immutable");
+		
 		try {
-			w.setWindowSize(new Dimension(800, 600));
+			window.setWindowSize(new Dimension(800, 600));
```

Comment:
as per coderabbitai review below, exception should fail the test, not ignore silently (for e.g, fail(e.getMessage()))